packages/synapse-react-client/src/components/widgets/query-filter/CombinedRangeFacetFilter.tsx

```diff
@@ -0,0 +1,157 @@
+  onChange,
+  collapsed = false,
+}: CombinedRangeFacetFilterProps) => {
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(collapsed)
+  const {
+    columnMin: col1Min,
+    // columnMax: col1Max  // not used
+    selectedMin: col1SelectedMin,
+    selectedMax: col1SelectedMax,
+  } = facetResults[0]
+  const {
+    //columnMin: col2Min,  // not used
+    columnMax: col2Max,
+    selectedMin: col2SelectedMin,
+    selectedMax: col2SelectedMax,
+  } = facetResults[1]
+
+  const isAnyValue =
+    !col1SelectedMin && !col1SelectedMax && !col2SelectedMin && !col2SelectedMax
+  const selectedMin = col2SelectedMin || col1Min
+  const selectedMax = col1SelectedMax || col2Max
+
+  const rangeType = columnType === 'DOUBLE' ? 'number' : 'date'
+
+  const handleRadioGroupChange = (
+    radioValue: RadioValuesEnum,
+    onChangeCallback: (
+      range: (RadioValuesEnum | number | string | undefined)[],
+    ) => void,
+  ) => {
+    setRadioValue(radioValue)
+
+    if (radioValue !== RadioValuesEnum.RANGE) {
+      onChangeCallback([radioValue, radioValue])
+    }
+  }
+
+  const [radioValue, setRadioValue] = useState(
+    getRadioValue(selectedMin, isAnyValue),
+  )
+
+  useEffect(() => {
+    setRadioValue(getRadioValue(selectedMin, isAnyValue))
+  }, [selectedMin, isAnyValue])
+
+  return (
+    <div>
+      <FacetFilterHeader
+        isCollapsed={isCollapsed}
+        label={label}
+        onClick={(isCollapsed: boolean) => setIsCollapsed(isCollapsed)}
+      ></FacetFilterHeader>
+      <Collapse in={!isCollapsed}>
+        <RadioGroup
+          value={radioValue}
+          id="combinedRangeSelector"
+          options={options}
+          onChange={(radioValue: string) =>
+            handleRadioGroupChange(radioValue as RadioValuesEnum, onChange)
+          }
+        ></RadioGroup>
+        {radioValue === RadioValuesEnum.RANGE &&
+          (col1Min === col2Max ? (
+            <label>{col2Max}</label>
+          ) : (
+            <>
+              {columnType === 'INTEGER' && (
+                <RangeSlider
+                  key="RangeSlider"
+                  domain={[col1Min, col2Max]}
+                  initialValues={{ min: selectedMin, max: selectedMax }}
+                  step={1}
+                  doUpdateOnApply={true}
+                  onChange={(values: RangeValues) =>
+                    onChange([col1Min, values.max, values.min, col2Max])
+                  }
+                >
+                  {'>'}
+                </RangeSlider>
+              )}
+              {columnType === 'DATE' && (
+                <Range
+                  key="Range"
+                  initialValues={{
+                    // From the backend, selectedMin is a formatted date (like "2021-06-15"), but columnMin is a unix timestamp in millis (like "1624651794856")
+                    min: col2SelectedMin ?? dayjs(parseInt(col1Min)).toString(),
+                    max: col1SelectedMax ?? dayjs(parseInt(col2Max)).toString(),
```

Comment:
Another bug fix made to both the RangeFacetFilter and this new version.

Reply:
Use the selected value (since it's a value that the calendar widget recognizes).  But if it's a unix timestamp (columnMin or columnMax), then parse it using dayjs for the range widget.

Reply:
Worth filing a PLFM bug? I guess it could be a breaking change so I dunno if they would fix it